src/libs/ReportUtils.js

```diff
@@ -373,6 +385,98 @@ function getIcons(report, personalDetails, policies, defaultIcon = null) {
     return _.map(sortedParticipants, item => item.avatar);
 }
 
+/**
+ * Get the displayName for a single report participant.
+ *
+ * @param {Object} participant
+ * @param {String} participant.displayName
+ * @param {String} participant.firstName
+ * @param {String} participant.login
+ * @param {Boolean} [useShortForm]
+ * @returns {String}
+ */
+function getDisplayNameForParticipant(participant, useShortForm = false) {
+    if (!participant) {
+        return '';
+    }
+
+    const loginWithoutSMSDomain = Str.removeSMSDomain(participant.login);
+    let longName = participant.displayName || loginWithoutSMSDomain;
+    if (Str.isSMSLogin(longName)) {
+        longName = LocalePhoneNumber.toLocalPhone(preferredLocale, longName);
+    }
+    const shortName = participant.firstName || longName;
+
+    return useShortForm ? shortName : longName;
+}
+
+/**
+ * @param {Object} participants
+ * @param {Boolean} isMultipleParticipantReport
+ * @returns {Array}
+ */
+function getDisplayNamesWithTooltips(participants, isMultipleParticipantReport) {
+    return _.map(
+        participants,
+        (participant) => {
+            const displayName = getDisplayNameForParticipant(participant, isMultipleParticipantReport);
+            const tooltip = Str.removeSMSDomain(participant.login);
+
+            let pronouns = participant.pronouns;
+            if (pronouns && pronouns.startsWith(CONST.PRONOUNS.PREFIX)) {
+                const pronounTranslationKey = pronouns.replace(CONST.PRONOUNS.PREFIX, '');
+                pronouns = Localize.translateLocal(`pronouns.${pronounTranslationKey}`);
+            }
+
+            return {
+                displayName,
+                tooltip,
+                pronouns,
+            };
+        },
+    );
+}
+
+/**
+ * Get the title for a report.
+ *
+ * @param {Object} report
+ * @param {Object} [personalDetailsForParticipants]
+ * @param {Object} [policies]
+ * @returns {String}
+ */
+function getTitle(report, personalDetailsForParticipants = {}, policies = {}) {
+    let title;
+    if (isChatRoom(report)) {
+        title = report.reportName;
+    }
+
+    if (isPolicyExpenseChat(report)) {
+        const reportOwnerPersonalDetails = lodashGet(personalDetailsForParticipants, report.ownerEmail);
+        const reportOwnerDisplayName = getDisplayNameForParticipant(reportOwnerPersonalDetails) || report.reportName;
+        title = report.isOwnPolicyExpenseChat ? getPolicyName(report, policies) : reportOwnerDisplayName;
+    }
+
+    if (isArchivedRoom(report)) {
+        title += ` (${Localize.translateLocal('common.archived')})`;
+    }
+
+    if (title) {
+        return title;
+    }
+
+    // Not a room or PolicyExpenseChat, generate title from participants
```

Comment:
Looks like `generate title from participants` happens below or is it somewhere else? What are we doing with this empty string?

Reply:
I think this thread might be outdated now, going to resolve.